### ISSUE TITLE
Global Content Blocks

### ISSUE THREAD

**Issue by marieepicraces** [realbig/RBM_TH_EPIC#136]
Sent via email but posting here as well:

I've tried several times to create Global Content Blocks using Joel's documentation with no success. Is there a step missing or a glitch somewhere?

**Comment by d4mation** [comment]
I believe it is a glitch. I am trying to figure it out now 👍

**Comment by marieepicraces** [comment]
Can we move this issue to the foreground? Creating Global Blocks will go a long way toward updating all of our events, once the glitch is resolved.
Thanks!

**Comment by d4mation** [comment]
Unfortunately, we had to open a ticket about this with a 3rd party and we're still waiting on them. I bumped the ticket with them asking for an update just now.

**Comment by marieepicraces** [comment]
Gotcha! Thanks for the update.

Marie Brooks

*Charity Champion & Registration Director*


p: 734.585.7101 ext. 708 a: 600 South Wagner, Ann Arbor, MI 48103

w: epicraces.com <https://www.epicraces.com/> e: marie@epicraces.com

**Comment by BigActual** [comment]
Followed up with the vendor this morning. Will keep you all posted as we make progress.

**Comment by BigActual** [comment]
From Elliot @ ACF:
Hi Steve

Thanks for the emails,
Elliot here - ACF dev.

Sorry for the delayed reply, It looks like your ticket got lost in the influx of ACF5 update tickets.

Thanks for the staging site.
I've logged into your site, and performed some testing, with some interesting results.

After testing the issue on your staging site, I exported your 2 x field groups (Event Single and Global Content Blocks) and imported them on my local site.
The strange thing is that on my site, the fields save correctly.

I had thought the issue may be due to the fact that you are cloning in a sub field for use as a parent field, so I setup some tests for this, but again, couldn't find any issues.

You linked to 2 x php files, however, I believe these are in a private repo and I was unable to see them. Perhaps you have some code in your theme that is either preventing the options value from saving, or modifying the flexible content field from loading? I'm not sure.

I encourage you to do as I have and export the field groups and import them into a local "vanilla" site for testing. Please see my attached screenshot of the field working as expected. Let me know what you find.

---------------------
@d4mation - Can you help Elliot with the 2 php files and recommendations?

**Comment by d4mation** [comment]
There has been progress with this. We now have the Global Content Blocks saving properly in my local test environment, but they are not loading into the Page Sections correctly. 

We will continue to keep you updated on this.

**Comment by marieepicraces** [comment]
Okay, great.

Marie Brooks

*Charity Champion & Registration Director*


p: 734.585.7101 ext. 708 a: 600 South Wagner, Ann Arbor, MI 48103

w: epicraces.com <https://www.epicraces.com/> e: marie@epicraces.com

**Comment by d4mation** [comment]
I pushed a fix for this to Live. The feature now works properly. Let me know if all looks good on your end 👍 

# There is a caveat to this

This was caused by what appears to be a bug in Advanced Custom Fields Pro regarding how it holds a cache for fields. It seems that the cache for Options Pages fields, or at least Clone fields on Options Pages, is broken and always returns a `false` value. This caused problems with the Event Edit Screen since as far as it was concerned, there were _never any saved Global Content Blocks_. With the Cache disabled for the Global Content Blocks, it was able to see them just fine.

To avoid waiting what could be weeks for a patch to the plugin from the vendor, I have manually applied the fix to Advanced Custom Fields Pro. This is normally something I'd never do (as if it were to receive an update  that didn't include the fix to this bug, then it will break again) but there was no way around it in this case if we wanted to get this up and running as fast as possible.

For future reference in case it becomes pertinent, I edited `advanced-custom-fields-pro/includes/api/api-value.php` in the `acf_get_value()` and `acf_format_value()` functions to not return the cached value if `$post_id == 'options'`.

This was necessary as of ACF Pro v5.7.6.

**Comment by marieepicraces** [comment]
Eric, I was finally able to create a global content block but I don't see it as an option to add to my private test page for Hot Cocoa Classic. Can you take a look at tell me what's up?

**Comment by d4mation** [comment]
@marieepicraces Did you scroll to the Content Blocks for your Page Section and add a Global Content Block? It seems to show your new Global Content Block that you made for me.

![image](https://i.imgur.com/wjiCxt4.gif)

**Comment by marieepicraces** [comment]
Yes, I did. Is there a time delay?? I can see Zero Waste now, but not the others that I've just created.

**Comment by d4mation** [comment]
After creating a new Global Content Block, are you refreshing the Edit Event page? It will only load new Global Content Blocks as choices when the Edit Page loads.

**Comment by marieepicraces** [comment]
I did just get the others to work. 

Will your work-around so that we can launch cause any difficulties down the road?

**Comment by d4mation** [comment]
It can. As I stated above, if Advanced Custom Fields Pro were to release an update that _does not_ include a fix for this bug, then it will stop the Global Content Blocks from working. This is unavoidable and there was literally no other way for me to address this aside from making you wait even longer.

I documented above what I did to correct the issue in the event that the plugin updates, the bug is not fixed, and the "patch" that I made needs to be re-applied.

I have made the author of the plugin aware of the issue and he hopefully will address this issue in the next update.
